adobe/react-spectrum
# fix(#3299): SearchAutocomplete/SearchWithin support icon prop override

Closes #3299 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding React Spectrum GitHub Issue #3299 .

## 📝 Test Instructions:

1. Open https://reactspectrum.blob.core.windows.net/reactspectrum/61069c227ee21cbb713035237b414e39ca813c09/storybook/index.html?path=/story/searchautocomplete--icon-filter&providerSwitcher-toastPosition=bottom
2. Verify the SearchAutocomplete uses the Filter icon.
3. Open https://reactspectrum.blob.core.windows.net/reactspectrum/61069c227ee21cbb713035237b414e39ca813c09/storybook/index.html?path=/story/searchautocomplete--icon-null&providerSwitcher-toastPosition=bottom
4. Verify the SearchAutocomplete does not render an icon.
5. Repeat steps 1-4 using a mobile user agent to verify that the behavior is similar with MobileSearchAutoComplete.
6. Open https://reactspectrum.blob.core.windows.net/reactspectrum/61069c227ee21cbb713035237b414e39ca813c09/storybook/index.html?path=/story/searchfield--icon-null&providerSwitcher-toastPosition=bottom
7. Verify the SearchField does not render an icon.
8. Open https://reactspectrum.blob.core.windows.net/reactspectrum/61069c227ee21cbb713035237b414e39ca813c09/storybook/index.html?path=/story/searchfield--isquiet-icon-null&providerSwitcher-toastPosition=bottom
9. Verify the quiet SearchField does not render an icon.
10. Open https://reactspectrum.blob.core.windows.net/reactspectrum/61069c227ee21cbb713035237b414e39ca813c09/storybook/index.html?path=/story/searchwithin--icon-filter&providerSwitcher-toastPosition=bottom
11. Verify the SearchWithin uses the Filter icon.
12. Open https://reactspectrum.blob.core.windows.net/reactspectrum/61069c227ee21cbb713035237b414e39ca813c09/storybook/index.html?path=/story/searchwithin--icon-null&providerSwitcher-toastPosition=bottom
13. Verify the SearchWithin does not render an icon.

## 🧢 Your Project:

Adobe/Accessibility
